src/components/contextual/pages/pool/PoolChart.vue

```diff
@@ -21,198 +23,309 @@ type Props = {
-    ? firstState.liquidity
-    : getPoolValue(firstState.amounts, firstState.prices);
-  const firstShares = firstState.totalShares;
-  const firstValuePerBpt = firstValue / firstShares;
-
-  return history.value.map(state => {
-    if (state.timestamp < firstState.timestamp) {
-      return 0;
+  const volumeData = periodSnapshots.map((snapshot, idx) => {
+    const value = parseFloat(snapshot.swapVolume);
+    let prevValue: number;
+
+    // get value of prev snapshot
+    if (idx === snapshotValues.value.length - 1) {
+      prevValue = 0;
+    } else if (idx === pariodLastSnapshotIdx) {
+      prevValue = parseFloat(snapshotValues.value[idx + 1].swapVolume);
+    } else {
+      prevValue = parseFloat(periodSnapshots[idx + 1].swapVolume);
     }
-
-    const currentValue = supportsPoolLiquidity.value
-      ? state.liquidity
-      : getPoolValue(state.amounts, state.prices);
-    const currentShares = state.totalShares;
-    const currentValuePerBpt = currentValue / currentShares;
-
-    return currentValuePerBpt / firstValuePerBpt - 1;
+    return {
+      name: timestamps.value[idx],
+      value: [timestamps.value[idx], value - prevValue]
+    };
   });
-});
 
-const series = computed(() => {
-  // TODO: currently HODL series is disabled when using pool liquidity
-  const supportsHODLSeries = !supportsPoolLiquidity.value;
+  const defaultHeaderStateValue =
+    Number(periodSnapshots[0].swapVolume) -
+    (isAllTimeSelected
+      ? 0
+      : Number(periodSnapshots[pariodLastSnapshotIdx].swapVolume));
+
+  return {
+    color: [tailwind.theme.colors.green['400']],
+    chartType: 'bar',
+    hoverColor: tailwind.theme.colors.pink['500'],
+    data: [
+      {
+        name: 'Volume',
+        values: volumeData
+      }
+    ],
+    defaultHeaderStateValue: fNum2(defaultHeaderStateValue, {
+      style: 'currency'
+    })
+  };
+});
 
-  const chartSeries = [
-    {
-      name: t('poolReturns'),
-      values: zip(timestamps.value, bptValues.value)
-    }
-  ];
+const defaultChartData = computed(() => {
+  const currentPeriodOption = periodOptions.value.find(
+    option => option.value === currentPeriod.value
+  );
+  let title = `${currentPeriodOption?.text} ${activeTab.value}`;
 
-  if (supportsHODLSeries) {
-    chartSeries.push({
-      name: 'HODL',
-      values: zip(timestamps.value, hodlValues.value)
-    });
+  if (activeTab.value === PoolChartTab.TVL) {
+    title = t('poolChart.defaultTitle.tvl');
   }
 
-  return chartSeries;
+  return { title, value: chartData.value.defaultHeaderStateValue };
 });
 
 /**
  * METHODS
  */
-function getPoolValue(amounts: string[], prices: number[]) {
-  return amounts
-    .map((amount, index) => {
-      const price = prices[index];
+function setCurrentPeriod(period: string) {
+  currentPeriod.value = Number(period);
+}
 
-      return price * parseFloat(amount);
-    })
-    .reduce((total, value) => total + value, 0);
+function setCurrentChartValue(value: {
+  value: [string, number];
```

Comment:
could have a more descriptive name so that you're not doing `value.value`. Also it could be an Object instead of array to describe what items 0 and 1 are.

Reply:
fixed, I changed the format